cbcontainers/state/components/sensor_daemon_set.go

```diff
@@ -408,7 +408,8 @@ func (obj *SensorDaemonSetK8sObject) mutateRuntimeEnvVars(container *coreV1.Cont
 
 	envVarBuilder := commonState.NewEnvVarBuilder().
 		WithCustom(customEnvs...).
-		WithSpec(sensorSpec.Env)
+		WithSpec(sensorSpec.Env).
+		WithProxySettings(agentSpec.Components.Settings.Proxy)
```

Comment:
For the **runtime sensor,** we want to unpack the `cbcontainers-runtime-resolver.cbcontainers-dataplane.svc.cluster.local` NO_PROXY value by default (unless it was already set by the customer). 

That is aim to simplify the customer CRD setting - customers was always forgetting to fill this value (documented [here](https://github.com/octarinesec/octarine-operator#using-http-proxy) ) or filled it incorrectly. That is why we wanted to fill them ourselves in the operator.